src/main/java/org/eclipse/basyx/submodel/metamodel/facade/SubmodelElementMapCollectionConverter.java

```diff
@@ -166,6 +176,30 @@ public static SubmodelElementCollection mapToSmECollection(Map<String, Object> s
 		return smElementsMap;
 	}
 	
+	/**
+	 * Checks if there are two elements in the provided Submodel Elements with the same IdShort <br>
+	 * 
+	 * @param Submodel Elements
+	 * @return True if there is duplicate IdShort present, otherwise False.
+	 */
+	@SuppressWarnings("unchecked")
+	private static boolean isDuplicateIdShortPresentInSubmodel(Object submodelElements) {
```

Comment:
This method should be named isDuplicateIdShortPresentInSubmodelElements. Additionally, please use a Map<String, Object> as input parameter and perform the cast on the caller's side

Reply:
- Rename it to isDuplicateIdShortPresentInSubmodelElements
- Performed the cast on the caller's side of isDuplicateIdShortPresentInSubmodelElements